### PR TITLE
Apply font names to gt_table class

### PR DESCRIPTION
**Pull request by Jeff-Thompson12**
Possibly solution to #185

### CODE REVIEW

**Comment by tdwils** [comment]
I tested it, and it works.

HTML in app in browser:

image

HTML download:

image

**Comment by Jeff-Thompson12** [comment]
@tdwils should I edit this PR so that the style is being assigned in the app itself? Ideally `{gt}` would make the fix, but it would give us the opportunity to allow the user to set the option.

**Comment by AARON-CLARK** [comment]
> @tdwils should I edit this PR so that the style is being assigned in the app itself? Ideally `{gt}` would make the fix, but it would give us the opportunity to allow the user to set the option.

@Jeff-Thompson12, yes! Let's do that. Also, @tdwils is OOO today.

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK Okay, there is an option to add additional CSS with the `gt` object itself. This should allow users to pass the desired fonts, but we will have to think about parsing the value ourselves.

File: R/mod_tableGen_utils.R [hunk 1]
```diff
@@ -614,15 +614,16 @@ create_gt_table <- function(data, input_table_title, input_table_footnote,
...
   data %>%
-    gt::gt(groupname_col = "ID") %>%
+    gt::gt(groupname_col = "ID", id = id) %>%
     gt::fmt_markdown(columns = c(Variable),
                      rows = stringr::str_detect(Variable,'&nbsp;') |
                        stringr::str_detect(Variable,'<b>') |
                        stringr::str_detect(Variable,'</b>')) %>%
     gt::tab_options(table.width = gt::px(700),
                     table.font.names = c("Times", "Arial"),
+                    table.additional_css = glue::glue('#{id} .gt_table {{font-family: Times, Arial;}}'),
```
Comment (AARON-CLARK) [inline R/mod_tableGen_utils.R:626]: So, I think this renders the `table.font.names = c("Times", "Arial"),` as redundant in `create_gt_table()`. On the flip side, should we update the downloaded R script to use this method for consistency sake? When I ran the script locally, I found that `table.additional_css = '.gt_table {font-family: Times, Arial;}'` worked. No need for random ID. Just thinking of the future when we allow the user to change the font.

**Review by AARON-CLARK — COMMENTED** [review]
@Jeff-Thompson12 looks great, just one comment!

File: R/mod_tableGen_utils.R (hunk 1 above)
Comment (Jeff-Thompson12) [reply R/mod_tableGen_utils.R:626]: > So, I think this renders the `table.font.names = c("Times", "Arial"),` as redundant in `create_gt_table()`.
Not quite redundant. I left both in place in the event that `{gt}` is doing some special processing on those values when creating the CSS. This ensures that the display will occur as expected when building tables locally.

> should we update the downloaded R script
Are we not calling `create_gt_table()` in the R script? I will look into this.

> No need for random ID. Just thinking of the future when we allow the user to change the font.
I was thinking of the future where we may want multiple tables on the same page. The random ID ensures that the formats are being applied to the appropriate tables. In the background `{gt}` is already randomly generating this value anyways. My code just allows us to have access to that value beforehand.

Comment (AARON-CLARK) [reply R/mod_tableGen_utils.R:626]: That all sounds good! I'll merge.

Yes, `create_gt_table()` is not exported. In the script, we just give the user the raw chunk of `gt` code which I think is actually really handy for (1) the user to customize and (2) development purposes.

**Pull request merged**